### PR TITLE
Feature/document/get all by role

### PR DESCRIPTION
**Pull request by TeeGoood**
### Type of Change

- [x] New features
- [ ] Bug fixes
- [ ] Breaking changes
- [ ] Refactor
- [ ] Documentation updates
- [ ] Other (please specify):

## Description
- get all documents by role API
- require role id in parameter
- can be filtered by organization, title, document type and response is paginated

### database
image

### sgcu_super admin role
image

### invalid role
image

### CODE REVIEW

**Review by wiraphatys — APPROVED** [review]
lgtm na krub bro.

**Comment by wiraphatys** [comment]
@bookpanda

File: internal/interface/handlers/document_handler.go [hunk 1]
```diff
@@ -103,7 +103,56 @@ func (h *DocumentHandler) GetDocumentByID(c *fiber.Ctx) error {
...
 func (h *DocumentHandler) GetDocumentsByRole(c *fiber.Ctx) error {
-	return nil
+	// validate parameter
+	getallDocumentsByRoleDTO := dtos.GetAllDocumentsByRoleDTO{
+		Page:         c.QueryInt("page", 1),
+		PageSize:     c.QueryInt("page_size", 10),
+		Title:        c.Query("title"),
+		DocumentType: c.Query("document_type"),
+		Organization: c.Query("organization"),
+		Role:         c.Params("role_id"),
```
Comment (bookpanda) [inline internal/interface/handlers/document_handler.go]: The role field is smt like `SCCU_ADMIN` right? Let's change it to `c.Params("role")` for more consistency. You can update the swagger by using `swag init -d ./internal/interface/handlers -g ../../../cmd/main.go -o ./docs -md ./docs/markdown --parseDependency --parseInternal` in root dir

File: internal/interface/handlers/document_handler.go [hunk 2]
```diff
@@ -103,7 +103,56 @@ func (h *DocumentHandler) GetDocumentByID(c *fiber.Ctx) error {
 // @Failure 500 {object} response.Response
 // @Router /documents/role/{role_id} [get]
```
Comment (bookpanda) [inline internal/interface/handlers/document_handler.go]: here

File: internal/interface/repositories/document_repository.go [hunk 3]
```diff
@@ -44,10 +44,12 @@ func (r *documentRepository) FindAllDocuments(args *FindAllDocumentsArgs) (*[]en
...
-		WHERE documents.type_id LIKE ?
+		WHERE users.role_id LIKE ?
+		AND  documents.type_id LIKE ?
 		AND	 LOWER(documents.title) LIKE ?
 		AND  documents.created_at BETWEEN ? AND ?
 		OFFSET ? LIMIT ?`,
+		fmt.Sprintf("%%%s%%", strings.ToUpper(args.Organization)),
 		fmt.Sprintf("%%%s%%", strings.ToUpper(args.DocumentType)),
 		fmt.Sprintf("%%%s%%", strings.ToLower(args.Title)),
 		args.StartTime,
```
Comment (bookpanda) [inline internal/interface/repositories/document_repository.go:55]: (FindAllDocuments)
seems like`args.Organization` is input as `WHERE users.role_id LIKE ?`, am i missing something?

Comment (bookpanda) [reply internal/interface/repositories/document_repository.go:55]: i see, understand now lmao

**Review by bookpanda — COMMENTED** [review]
lgtm

File: internal/interface/handlers/document_handler.go (hunk 1 above)
Comment (wiraphatys) [reply internal/interface/handlers/document_handler.go]: @TeeGoood did you update kub

Comment (TeeGoood) [reply internal/interface/handlers/document_handler.go]: updated

**Pull request merged**